provider/baked/src/binary_search.rs

```diff
@@ -0,0 +1,45 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+//! Data stored as slices, looked up with binary search
+
+#[cfg(feature = "export")]
+use databake::*;
+#[cfg(feature = "export")]
+use icu_provider::prelude::*;
+
+#[cfg(feature = "export")]
+pub fn bake(
```

Comment:
Thought: I'm happy that we can move more common runtime baked data code into this crate.